core/src/mos6502.h

```diff
@@ -80,12 +84,14 @@ class Mos6502 : public ICpu {
     Pipeline create_load_instruction(Opcode opcode);
     Pipeline create_compare_instruction(Opcode opcode);
 
-    Pipeline create_addressing_steps(AddressMode address_mode);
+    Pipeline create_addressing_steps(AddressMode address_mode,
+            bool is_write = false);
```

Comment:
Not sure what makes most sense here: default to write, default to read, or require it to be specified (in that case it should perhaps belong in the opcode struct?).

Reply:
I think this is good for now. We can change it in a separate PR if we come up with something nicer.